src/io/TXTTableWriter.cpp

```diff
@@ -30,16 +30,21 @@ void TXTTableWriter::addData(
   data.name = name;
   data.type = type;
   _data.push_back(data);
+
+  std::string delimiter = _data.empty() ? "" : "  ";
```

Comment:
`delimiter` is a tricky choice as the space separating vectorial components is also a delimiter.

I would handle this edge case here and concentrate on just the data section below.
```
  if (!_data.empty()) {
    _outputStream << ' ';
  }
```

Also, this has to move above `_data.push_back()`.

Reply:
> delimiter is a tricky choice as the space separating vectorial components is also a delimiter.

I don't see the trickyness here. What exactly is the problem with that?

> Also, this has to move above _data.push_back().

Yes, that's what happens in last minute changes :see_no_evil:

Reply:
The point I am trying to make, is that we can simplify and clarify this.
fmtlibs makes a lot of fiddly string handling very clean.